syntax='proto3';

service AuthService{
    rpc SignUp(SignUpRequest) returns (SignUpResponse);
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc RefreshToken(RefreshRequest) returns (RefreshResponse);
    rpc Logout(LogoutRequest) returns (LogoutResponse);
}

message SignUpRequest {
    string username = 1;
    string email = 2;
    string password = 3;
    string bio = 4;
}

message SignUpResponse{
    string access_token = 1;
    string refresh_token = 2;
    string token_type = 3;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  string access_token = 1;
  string refresh_token = 2;
  string token_type = 3;
}

message RefreshRequest {
  Token token = 1;
}

message RefreshResponse {
  string access_token = 1;
  string refresh_token = 2;
  string token_type = 3;
}

message LogoutRequest {
  Token token = 1;
}

message LogoutResponse {
  string message = 1;
}

message Token {
  string jwt_string = 1;
  string user_id = 2;
  string session_id = 3;
  int64 exp = 4;
}

